appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecScanEventListener.java

```diff
@@ -0,0 +1,29 @@
+/*-
+ * Copyright (C) 2023 Vaadin Ltd
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ *
+ * See <https://vaadin.com/commercial-license-and-service-terms> for the full license.
+ */
+package com.vaadin.appsec.backend;
+
+import java.io.Serializable;
+import java.util.EventListener;
+
+/**
+ * A listener of {@link AppSecScanEvent} that can be added to
+ * {@link AppSecService} with
+ * {@link AppSecService#addScanEventListener(AppSecScanEventListener)}.
+ */
+@FunctionalInterface
+public interface AppSecScanEventListener extends EventListener, Serializable {
+
+    /**
+     * Operation performed when a new scan event has completed.
```

Comment:
We also need some way to inform the appsec ui part about unhandled vulnerabilities at the moment when the developer opens their application ui but the appseckit is not yet open. So we either need some kind of event mechanism for this purpose, or we could have e.g. `shouldShowNotification()` or similar method in `AppSecService` that could be called upon new application ui init.

Reply:
True, now we don't have a specific event for this. Let's add this on another PR, I've created #39 to track this.

Reply:
Great, thanks! I'll close the comment then.